stytch/src/main/kotlin/com/stytch/java/common/ParseJWTClaims.kt

```diff
@@ -0,0 +1,47 @@
+package com.stytch.java.common
+
+import org.jose4j.jwk.HttpsJwks
+import org.jose4j.jwt.JwtClaims
+import org.jose4j.jwt.consumer.JwtConsumerBuilder
+import java.util.Date
+
+internal data class ParsedJWTClaims(
+    val payload: JwtClaims,
+    val customClaims: Map<String, Any> = emptyMap(),
+)
+
+internal val IGNORED_CLAIMS = listOf("aud", "exp", "iat", "iss", "jti", "nbf", "sub")
+
+internal fun parseJWTClaims(
+    jwt: String,
+    jwtOptions: JwtOptions,
+    jwksClient: HttpsJwks,
+    options: ParseJWTClaimsOptions,
+): ParsedJWTClaims {
+    val now = Date().time
+    val jwtConsumer = JwtConsumerBuilder().apply {
+        setRequireExpirationTime()
+        setAllowedClockSkewInSeconds(options.leeway)
+        setRequireSubject()
+        setExpectedAudience(jwtOptions.audience)
+        setExpectedIssuer(jwtOptions.issuer)
+        setVerificationKey(jwksClient.jsonWebKeys[0].key)
```

Comment:
Does this mean we only take the first key of the JWK? That's a pretty fragile assumption - any active JWK can be used to sign a JWT. 

Can we use `setVerificationKeyResolver` instead with either 
https://javadoc.io/doc/org.bitbucket.b_c/jose4j/0.5.1/org/jose4j/keys/resolvers/JwksVerificationKeyResolver.html
or 
https://javadoc.io/doc/org.bitbucket.b_c/jose4j/0.5.1/org/jose4j/keys/resolvers/HttpsJwksVerificationKeyResolver.html